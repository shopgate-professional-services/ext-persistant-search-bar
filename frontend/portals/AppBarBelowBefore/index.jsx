import React from 'react';
import isIOSTheme from '@shopgate-ps/pwa-extension-kit/env/helpers/isIOSTheme';
import SearchField from '../../components/SearchField';

jest.mock('../../selectors', () => ({
  isSearchBarVisible: jest.fn().mockReturnValue(true),
}));

/**
 * Renders SeachedField component in app-bar.below.before portal
 * @returns {JSX}
 */
const AppBarBelowBefore = () => (
  <SearchField isIOSTheme={isIOSTheme} />
);

export default AppBarBelowBefore;
