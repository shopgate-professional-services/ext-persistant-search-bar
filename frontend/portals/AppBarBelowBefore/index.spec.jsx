import React from 'react';
import { shallow } from 'enzyme';
import isIOSTheme from '@shopgate-ps/pwa-extension-kit/env/helpers/isIOSTheme';
import AppBarBelowBefore from './index';

jest.mock('@shopgate/engage/core', () => ({
  useRoute: () => ({ id: 123 }),
}));

jest.mock('@shopgate/pwa-ui-shared/Glow', () => <div>Glow</div>);

jest.mock('../../components/SearchField', () => <div>SearchField</div>);

jest.mock('@shopgate-ps/pwa-extension-kit/env/helpers/isIOSTheme', () => () => true);

describe('<AppBarBelowBefore />', () => {
  it('should render', () => {
    const component = shallow(<AppBarBelowBefore isIOSTheme={isIOSTheme} />);
    expect(component).toMatchSnapshot();
  });
});
