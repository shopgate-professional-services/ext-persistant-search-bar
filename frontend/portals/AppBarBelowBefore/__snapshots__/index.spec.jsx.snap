// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBarBelowBefore /> should render 1`] = `
<[object Object]
  isIOSTheme={[Function]}
  pageId={123}
/>
`;
